### PR TITLE
Add doc for training a policy

### PR DESCRIPTION
**Pull request by yeukfu**
[#1530](https://github.com/rlworkgroup/garage/issues/1530).

### CODE REVIEW

File: docs/user/training_a_policy.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
+# Training a policy to solve an environment
+This page will guide you training a policy to solve an environment.
+### Define an experiment
+In garage, we train a policy in an experiment, which is a function wrapped by a decorator called `wrap_experiment`. Below is an simple example. `wrap_experiment` could have some arguments. You can see the [experiments doc](https://garage.readthedocs.io/en/latest/user/experiments.html) for details of running experiments.
```
Comment (ryanjulian) [inline docs/user/training_a_policy.md:4]: i think you need to use a relative link here. you don't know which version of the docs the user is reading.

See https://github.com/readthedocs/recommonmark/issues/108

File: docs/user/training_a_policy.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    with LocalTFRunner(ctxt) as runner:
+        ...
+```
+### Construct an environment
+Garage supports many envinronments. You can also implement your own environment like [this](https://garage.readthedocs.io/en/latest/user/implement_env.html). In this example, we choose `CartPole-V1` environment.
+```py
+env = GarageEnv(env_name='CartPole-v1')
+```
+### Construct a policy and an algorithm
+Construct your policy and choose an algorithm to train it. Here, we use `CategoricalMLPPolicy` and `TRPO`, you can also implement your own algorithm like [this](https://garage.readthedocs.io/en/latest/user/implement_algo.html).
```
Comment (ryanjulian) [inline docs/user/training_a_policy.md:33]: Relative link https://github.com/readthedocs/recommonmark/issues/108

File: docs/user/training_a_policy.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    set_seed(seed)
+    # PyTorch
+    runner = LocalRunner(ctxt)
+    ...
+    # TensorFlow
+    with LocalTFRunner(ctxt) as runner:
+        ...
+```
+### Construct an environment
+Garage supports many envinronments. You can also implement your own environment like [this](https://garage.readthedocs.io/en/latest/user/implement_env.html). In this example, we choose `CartPole-V1` environment.
```
Comment (ryanjulian) [inline docs/user/training_a_policy.md:28]: Relative link

File: docs/user/training_a_policy.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+            max_kl_step=0.01)
+```
+### Train your policy
+The final step is calling `runner.setup` and `runnner.train` to co-ordinate training the policy.
+```py
+runner.setup(algo, env)
+runner.train(n_epochs=100, batch_size=4000)
+```
+### Run the experiment
+In the above steps, we construct the requited componects to train a `CategoricalMLPPolicy` with `TRPO` to solve `CartPole-v1` and wrap all into an experiment function. You can find the full example in `examples/tf/trpo_cartpole.py`, which is also pasted below:
```
Comment (ryanjulian) [inline docs/user/training_a_policy.md:55]: link to the code?

File: docs/user/training_a_policy.md (hunk 1 above)
Comment (yeukfu) [reply docs/user/training_a_policy.md:4]: Good idea. Will do.

**Comment by ryanjulian** [comment]
I think you need to add this page to `docs/index.rst` for it to appear in the Table of Contents

File: docs/user/training_a_policy.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    set_seed(seed)
+    # PyTorch
+    runner = LocalRunner(ctxt)
+    ...
+    # TensorFlow
+    with LocalTFRunner(ctxt) as runner:
+        ...
+```
+### Construct an environment
+Garage supports many envinronments. You can also implement your own environment like [this](implement_env.rst). In this example, we choose `CartPole-V1` environment.
```
Comment (maliesa96) [inline docs/user/training_a_policy.md:28]: typo (envinronments)

File: docs/user/training_a_policy.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+            max_kl_step=0.01)
+```
+### Train your policy
+The final step is calling `runner.setup` and `runnner.train` to co-ordinate training the policy.
+```py
+runner.setup(algo, env)
+runner.train(n_epochs=100, batch_size=4000)
+```
+### Run the experiment
+In the above steps, we construct the requited componects to train a `CategoricalMLPPolicy` with `TRPO` to solve `CartPole-v1` and wrap all into an experiment function. You can find the full example in [`examples/tf/trpo_cartpole.py`](https://github.com/rlworkgroup/garage/blob/master/examples/tf/trpo_cartpole.py), which is also pasted below:
```
Comment (maliesa96) [inline docs/user/training_a_policy.md:55]: typo (components)

File: docs/user/training_a_policy.md [hunk 7]
```diff
@@ -0,0 +1,145 @@
+# Training a policy to solve an environment
+This page will guide you training a policy to solve an environment.
+### Define an experiment
+In garage, we train a policy in an experiment, which is a function wrapped by a decorator called `wrap_experiment`. Below is an simple example. `wrap_experiment` could have some arguments. You can see the [experiments doc](experiments.rst) for details of running experiments.
```
Comment (maliesa96) [inline docs/user/training_a_policy.md:4]: You should probably mention somewhere here that users should pick a policy that is compatible with the environment's observations and actions (CNN for image observations, discrete policy for discrete action spaces, etc)

Comment (yeukfu) [reply docs/user/training_a_policy.md:4]: Good point. Will do.

**Comment by ryanjulian** [comment]
Don't forget to add an acknowledgement for yourself at the bottom of the page! See https://github.com/rlworkgroup/garage/pull/1638 for an example

**Comment by yeukfu** [comment]
> Don't forget to add an acknowledgement for yourself at the bottom of the page! See #1638 for an example

Got it

File: docs/user/benchmarking.md [hunk 8]
```diff
@@ -20,7 +20,7 @@ Use `@wrap_experiment` to define your experiment. `ctxt` is used for
 The second and third parameters of the function should be `env_id` and `seed`.
 You should give your function a good name because it is used as the label name
 when it comes to plotting. See more [examples](https://github.com/rlworkgroup/garage/tree/master/benchmarks/src/garage_benchmarks/experiments).
-You can see detailed documentations of defining experiments [here](experiments.rst).
+You can see detailed documentations of defining experiments [here](experiments).
```
Comment (ahtsan) [inline docs/user/benchmarking.md:23]: The link doesn't work. You should probably keep `experiments.rst`?

File: docs/user/benchmarking.md [hunk 9]
```diff
@@ -5,7 +5,7 @@ This guide explains how to write and run garage benchmark scripts.
 ## Prerequisites
 
 Firstly, you should have installed garage. If not, you can see instructions in
-the [latest documentation](installation.rst) to install garage.
+the [latest documentation](installation) to install garage.
```
Comment (ahtsan) [inline docs/user/benchmarking.md:8]: The link doesn't work. You should probably keep `installation.rst`?

File: docs/user/training_a_policy.md [hunk 10]
```diff
@@ -0,0 +1,189 @@
+# Training a policy to solve an environment
+
+This page will guide you training a policy to solve an environment.
+
+## Define an experiment
+
+In garage, we train a policy in an experiment, which is a function wrapped by a
+decorator called `wrap_experiment`. Below is an simple example.
+`wrap_experiment` could have some arguments. You can see the [experiments doc](experiments)
```
Comment (ahtsan) [inline docs/user/training_a_policy.md:9]: The link doesn't work. You should probably keep `experiments.rst`?

File: docs/user/training_a_policy.md [hunk 11]
```diff
@@ -0,0 +1,189 @@
...
+
+```py
+env = GarageEnv(env_name='CartPole-v1')
+```
+
+## Construct a policy and an algorithm
+
+Construct your policy and choose an algorithm to train it. Here, we use
+`CategoricalMLPPolicy` and `TRPO`, you can also implement your own algorithm
+like [this](implement_algo). Your policy should be compatible with the
```
Comment (ahtsan) [inline docs/user/training_a_policy.md:57]: The link doesn't work. You should probably keep `implement_algo.rst`?

File: docs/user/training_a_policy.md [hunk 12]
```diff
@@ -0,0 +1,189 @@
...
+    ...
+    # TensorFlow
+    with LocalTFRunner(ctxt) as runner:
+        ...
+```
+
+## Construct an environment
+
+Garage supports many environments. You can also implement your own environment
+like [this](implement_env). In this example, we choose `CartPole-V1`
```
Comment (ahtsan) [inline docs/user/training_a_policy.md:46]: The link doesn't work. You should probably keep `implement_env.rst`?

File: docs/user/training_a_policy.md [hunk 13]
```diff
@@ -0,0 +1,189 @@
...
+
+## Run the experiment
+
+In the above steps, we construct the required components to train a
+`CategoricalMLPPolicy` with `TRPO` to solve `CartPole-v1` and wrap all into an
+experiment function. You can find the full example in [`examples/tf/trpo_cartpole.py`](https://github.com/rlworkgroup/garage/blob/master/examples/tf/trpo_cartpole.py),
+which is also pasted below:
+
+```py
+from garage import wrap_experiment
```
Comment (ahtsan) [inline docs/user/training_a_policy.md:94]: I personally prefer you to show the full snippet first, and then explained each part.

File: docs/user/benchmarking.md (hunk 8 above)
Comment (yeukfu) [reply docs/user/benchmarking.md:23]: The master branch uses `experiments.rst` and it doesn't work. In the doc preview of this PR, I test it and it works.

Comment (ahtsan) [reply docs/user/benchmarking.md:23]: I see, thanks for clarifying.

File: docs/user/training_a_policy.md [hunk 14]
```diff
@@ -0,0 +1,189 @@
...
+
+```py
+@wrap_experiment
+def my_first_experiment():
+    ...
+```
+
+## Construct a LocalRunner
+
+Within the experiment, we need a `LocalRunner` to sets up important state (such
```
Comment (irisliucy) [inline docs/user/training_a_policy.md:20]: 'Within the experiment, we need a `LocalRunner` to set up...'

File: docs/user/training_a_policy.md [hunk 15]
```diff
@@ -0,0 +1,189 @@
+# Training a policy to solve an environment
+
+This page will guide you training a policy to solve an environment.
```
Comment (irisliucy) [inline docs/user/training_a_policy.md:3]: 'This page will guide you on how to a policy to solve an environment.'

File: docs/user/training_a_policy.md [hunk 16]
```diff
@@ -0,0 +1,189 @@
...
+
+algo = TRPO(env_spec=env.spec,
+            policy=policy,
+            baseline=baseline,
+            max_path_length=100,
+            discount=0.99,
+            max_kl_step=0.01)
+```
+
+## Train your policy
```
Comment (irisliucy) [inline docs/user/training_a_policy.md:76]: I think it's better to use a general, objective sub-header like 'Train the policy'.

File: docs/user/training_a_policy.md [hunk 17]
```diff
@@ -0,0 +1,189 @@
...
+env = GarageEnv(env_name='CartPole-v1')
+```
+
+## Construct a policy and an algorithm
+
+Construct your policy and choose an algorithm to train it. Here, we use
+`CategoricalMLPPolicy` and `TRPO`, you can also implement your own algorithm
+like [this](implement_algo). Your policy should be compatible with the
+environment's observations and action space (CNN for image observations,
+discrete policy for discrete action spaces, etc).
```
Comment (irisliucy) [inline docs/user/training_a_policy.md:59]: Good point on reminding reader about compatible observations and action space. Maybe, you can also provide an intuition about why we use `CategoricalMLPPolicy` in the `CartPole-v1` environment using `TRPO` algorithm.

Comment (yeukfu) [reply docs/user/training_a_policy.md:59]: Good idea
